app/src/main/java/com/github/se/polyfit/model/post/Post.kt

```diff
@@ -70,6 +71,33 @@ data class Post(
       }
     }
 
+    fun deserialize(data: Map<String, Any?>): Post? {
+      return try {
+        val userId = data["userId"] as String
+
+        val description = data["description"] as? String ?: ""
+        val location = Location.deserialize(data["location"] as Map<String, Any>)
+        val meal = Meal.deserialize(data["meal"] as Map<String, Any>)
+        val createdAt = deserializeLocalDate(data, "createdAt") ?: LocalDate.now()
+
+        val newPost = Post(userId, description, location, meal, createdAt)
+
+        newPost
+      } catch (e: Exception) {
+        Log.e("Post", "Failed to deserialize Post object: ${e.message}", e)
+        throw IllegalArgumentException("Failed to deserialize Post object", e)
+      }
+    }
+
+    private fun deserializeLocalDate(data: Map<String, Any?>, key: String): LocalDate? {
+      return try {
+        val dateString = data[key] as? String
+        dateString?.let { LocalDate.parse(it) }
```

Comment:
It is better to fail and go to the catch then return a null value.

Reply:
I completely missed that thank you for catching it.